### DISCUSSION TITLE
I made a colors-browser for Flet’s community.

### DISCUSSION THREAD

**Discussion by ndonkoHenri** [flet-dev/flet#373] [Show and tell]
We all know that working with colors is never easy(especially for the non-designers). To able to choose a color for our UI, we need to be able to visualize all those available. I then decided to refactor/rebuild the icons-browser and made a colors-browser out of it. Equally deployed this on fly.io. [flet-colors-browser.fly.dev](https://flet-colors-browser.fly.dev/)

Here is a link to my Repo
[ndonkoHenri/Flet-Color-Browser](https://github.com/ndonkoHenri/Flet-Color-Browser)

I also made a pull request to Flet's _example/python/community_ in order to have it added to 'community built stuffs' [https://github.com/flet-dev/examples/pull/29](https://github.com/flet-dev/examples/pull/29)

Let me have your impressions on this please.
Thanks.

**Comment by FeodorFitsner** [comment]
Thanks for making the app online.

Could you please elaborate on how to use your app? I was expecting to see something like https://gallery.flutter.dev/#/demo/colors, but you have a different approach.

**Reply by ndonkoHenri** [reply]
> Could you please elaborate on how to use your app?

Just like the icons-browser you made.
In the TextField, the developer enters a string(eventually that if the color he wants), then validâtes by pressing «enter» or the search button, and gets in a Grid, a list of colors containing the entered string/search-term.

This grid of colors shows exactly how each color looks, and hence helps the developer better choose the one he is looking for, or the one best for his project.
I also added a button to switch the Theme Mode, because not all colors could easily be seen/identified in the 'light' or 'dark' mode.

With the help of Appveyor, I made [releases](https://github.com/ndonkoHenri/Flet-Color-Browser/releases) for the Three main OSs(Win,Mac,Linux).

**Reply by FeodorFitsner** [reply]
Maybe there is a bug, but this is what I'm getting while searching for `gree`:

![image](https://user-images.githubusercontent.com/5041459/191873262-22bbfe82-4d08-43c7-beed-97d4216d86ca.png)

**Reply by ndonkoHenri** [reply]
No it's not a bug.
All the results for your search term have "gree" somewhere in it. (32 found)

Ex: if you search term is "a", it returns all the colors containing "a" in it. (orange,black,...) and not just those starting with "a".

**Reply by FeodorFitsner** [reply]
Right, but why only "green" is green, i.e. has colour?

**Reply by ndonkoHenri** [reply]
Ah okkk. I see what you mean now.

Let me have a look at it right now.
Thanks.

**Reply by ndonkoHenri** [reply]
Thanks for testing it @FeodorFitsner .
I saw my error and had it fixed.

I also took advantage of this to add more comment in my code..

Have a look at it now and let me know please. (From The website, or releases)

**Reply by FeodorFitsner** [reply]
Yep, it works now. Thanks for the fix!

However, it seems to me that it would be more beautiful (logical?) to list found colours in ListView, one-under-one, rather than as "folders" in a GridView. Don't you think so?

![image](https://user-images.githubusercontent.com/5041459/191879681-5ff0ad79-6f61-42e9-9c54-d0b2f20d0405.png)

**Reply by ndonkoHenri** [reply]
Interesting idea. The colors will be more visible.

But what about the navigation between the main colors? (Tabs? As in the Flutter example?) 🤔

**Reply by FeodorFitsner** [reply]
Why not?

**Reply by ndonkoHenri** [reply]
Ok.
And...what about the Hex colors? Any existing utils in Flet to have the Hex values of the colors found in colors.py module?

**Reply by ndonkoHenri** [reply]
> Yep, it works now. Thanks for the fix!
> 
> 
> 
> However, it seems to me that it would be more beautiful (logical?) to list found colours in ListView, one-under-one, rather than as "folders" in a GridView. Don't you think so?

> ![image](https://user-images.githubusercontent.com/5041459/191879681-5ff0ad79-6f61-42e9-9c54-d0b2f20d0405.png)


I forgot to mention in this discussion that I had this done.

https://github.com/ndonkoHenri/Flet-Color-Browser

**Comment by hololeo** [comment]
Really great stuff [ndonkoHenri](https://github.com/ndonkoHenri)!

**Reply by ndonkoHenri** [reply]
Thanks 🙏🏽
